keras/engine/base_layer.py

```diff
@@ -325,6 +325,8 @@ def __init__(self,
     # Mutable properties
     # Indicates whether the layer's weights are updated during training
     # and whether the layer's updates are run during training.
+    if not isinstance(trainable, bool):
```

Comment:
The `trainable` argument should also support a scalar boolean `tf.Variable`.

In addition, there is a risk here that some people might be passing `0` or `1` and would then see their code break. We would need to run a global TAP to check for such cases.